core/node/eth_sender/src/eth_tx_aggregator.rs

```diff
@@ -471,8 +583,32 @@ impl EthTxAggregator {
                 );
                 return Ok(());
             }
+
+            if Self::is_pending_gateway_upgrade(storage, chain_protocol_version_id).await
+                && agg_op.is_execute()
+            {
```

Comment:
Current implementation not only stops aggregating execute operations but also all operations that come after executes in the queue. There can be proof operations for old batches so it won't work. Let's move this to the place where `get_next_ready_operation` is called